source/hwIo/ioThread.py

```diff
@@ -197,7 +197,10 @@ def fakeApc(param):
 		self.handle = None
 
 	def run(self):
-		while True:
-			ctypes.windll.kernel32.SleepEx(winKernel.INFINITE, True)
-			if self.exit:
-				break
+		try:
+			while True:
+				ctypes.windll.kernel32.SleepEx(winKernel.INFINITE, True)
+				if self.exit:
+					break
+		except Exception:
+			log.critical("Exception in IoThread function", exc_info=True, stack_info=True)
```

Comment:
should we reraise the exception afterwards?
I'm concerned about change of behavior

Reply:
The behavior isn't changed:

### Old behavior
1. An access violation occurs in the IoThread
2. As it is unhandled, it logs to stderr
3. It ends up as unhandled exception in the log
4. As there is a crash in the run function of the thread, the run function ends and the thread is no longer running.

### New behavior
1. An access violation occurs within the try block in the IoThread
2. As it is handled, it logs on the critical level and the stacks of all threads are dumped to the log as well
3. As the try/except is placed around the while loop, the run function exits gracefully and the thread shuts down as well